File: 프로그래머스/2/84512. 모음 사전/모음 사전.py
def solution(word):
    answer = 0
    dict_w={'A':1, 'E':2, 'I':3, 'O':4, 'U':5}
    # [0,5,25,125,625]
    scale = [780, 155, 30, 5, 0]
    word=word.ljust(5," ")
    for i in range(len(word)):
        if word[i]==" ":
            continue
        answer+=scale[i]*(dict_w[word[i]]-1)+dict_w[word[i]]
    
    return answer

"""
이해를  위한 예시
A
    AA
        AAA

            AAAA
                AAAAA
                AAAAE
                AAAAI
                AAAAO
                AAAAU

            AAAE
                AAAEA
                AAAEE
                AAAEI
                AAAEO
                AAAEU

            AAAI
                AAAIA
                AAAIE
                AAAII
                AAAIO
                AAAIU

            AAAO
                AAAOA
                AAAOE
                AAAOI
                AAAOO
                AAAOU

            AAAU
                AAAUA
                AAAUE
                AAAUI
                AAAUO
                AAAUU

        AAE
            AAEA
            AAEE
            AAEI
            AAEO
            AAEU
...

"""

File: 프로그래머스/2/84512. 모음 사전/모음 사전.java
// 'A', 'E', 'I', 'O', 'U'

import java.util.*;
class Solution {
    public int solution(String word) {
        int answer = 0;
        Map<Character, Integer> map = new HashMap<>();
        map.put('A', 1);
        map.put('E', 2);
        map.put('I', 3);
        map.put('O', 4);
        map.put('U', 5);
        
        int[] scale = {780, 155, 30, 5, 0};

        for (int i = 0; i < word.length(); i++) {
            char c = word.charAt(i);
            answer += scale[i] * (map.get(c) - 1) + map.get(c);
        }
            
return answer;
    }
}